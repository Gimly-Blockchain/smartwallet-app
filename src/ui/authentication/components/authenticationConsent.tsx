import React from 'react'
import { ButtonSection } from 'src/ui/structure/buttonSectionBottom'
import { Text, StyleSheet, View } from 'react-native'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Typography, Spacing } from 'src/styles'
import { IssuerCard } from '../../documents/components/issuerCard'
import { AuthenticationRequestSummary } from '../../../actions/sso/types'

interface Props {
  authenticationDetails: AuthenticationRequestSummary
  generateAndSendAuthenticationResponse: Function
  cancelAuthenticationRequest: Function
}

interface State {}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.backgroundLightMain,
  },
  topSection: {
    flex: 0.9,
  },
  issuerCard: {
    marginTop: Spacing.LG,
  },
  authRequestContainer: {
    flex: 1,
    paddingHorizontal: '10%',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginTop: Spacing.XL,
  },
  authRequestText: {
    ...Typography.baseFontStyles,
    ...Typography.centeredText,
    fontSize: Typography.textLG,
    marginTop: Spacing.SM,
  },
  buttonSection: {
    flex: 0.1,
  },
})

export class AuthenticationConsentComponent extends React.Component<
  Props,
  State
> {
  public state = {
    pending: false,
  }

  private handleConfirm = () => {
    this.setState({ pending: true })
    return this.props.generateAndSendAuthenticationResponse()
  }

  public render() {
    const { requester, description } = this.props.authenticationDetails
    return (
      <View style={styles.container}>
        <View style={styles.topSection}>
          <IssuerCard issuer={requester} style={styles.issuerCard} />
          <View style={styles.authRequestContainer}>
            <Text style={styles.authRequestText}>
              {I18n.t(strings.WOULD_YOU_LIKE_TO)}
            </Text>
            <Text
              style={[styles.authRequestText, { fontSize: Typography.text4XL }]}
            >
              {description}
            </Text>
            <Text style={styles.authRequestText}>
              {I18n.t(strings.WITH_YOUR_SMARTWALLET)}
            </Text>
          </View>
        </View>
        <View style={styles.buttonSection}>
          <ButtonSection
            disabled={this.state.pending}
            denyDisabled={this.state.pending}
            confirmText={I18n.t(strings.AUTHORIZE)}
            denyText={I18n.t(strings.DENY)}
            handleConfirm={this.handleConfirm}
            handleDeny={() => this.props.cancelAuthenticationRequest()}
          />
        </View>
      </View>
    )
  }
}
