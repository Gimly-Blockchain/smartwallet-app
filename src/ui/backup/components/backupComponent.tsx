import * as React from 'react'
import { ScrollView, StyleSheet, Text, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { fontMain } from '../../../styles/typography'
import { GradientButton } from '../../structure/gradientButton'
import { NavigationSection } from '../../structure/navigationSection'

interface Props {
  isLoading: boolean
  isAutoBackupEnabled: boolean
  enableAutoBackup: () => void
  onDisableAutoBackup: () => void
  exportBackup: () => void
  goBack: () => void
  lastBackup?: string
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  scrollView: {
    padding: 18,
  },
  section: {
    backgroundColor: Colors.black,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    marginTop: 32,
    borderRadius: 8,
    padding: 24,
    paddingBottom: 28,
  },
  title: {
    fontFamily: fontMain,
    color: Colors.white,
    fontSize: 28,
  },
  subtitle: {
    fontFamily: fontMain,
    color: Colors.white,
    fontSize: 22,
    textAlign: 'center',
  },
  description: {
    color: Colors.white050,
    fontSize: 16,
    marginTop: 8,
    textAlign: 'center',
  },
  warning: {
    color: Colors.error,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 28,
  },
  info: {
    textAlign: 'center',
    color: Colors.purpleMain,
    marginTop: 28,
  },
  buttonContainer: {
    marginTop: 22,
    width: '100%',
    paddingVertical: 9,
    margin: 0,
    borderRadius: 8,
  },
  buttonText: {
    fontSize: 18,
  },
})
const BackupComponent: React.FC<Props> = ({
  isLoading,
  enableAutoBackup,
  onDisableAutoBackup,
  exportBackup,
  isAutoBackupEnabled,
  goBack,
  lastBackup,
}) => {
  return (
    <Container style={styles.container}>
      <NavigationSection
        onNavigation={goBack}
        isBackButton={true}
        isDark={true}
      />
      <ScrollView style={styles.scrollView}>
        <Text style={styles.title}>Backup options</Text>

        <View style={styles.section}>
          <Text style={styles.subtitle}>Backup data on your own</Text>
          <Text style={styles.description}>
            Download an encrypted copy of the data in your SmartWallet to your
            device. Make sure to keep your backups up-to-date and stored
            somewhere safe
          </Text>
          <GradientButton
            containerStyle={styles.buttonContainer}
            textStyle={styles.buttonText}
            onPress={exportBackup}
            text={'Export file'}
            disabled={isLoading}
          />
          {lastBackup && (
            <Text style={styles.info}>
              {`Last backup ${new Date(lastBackup).toLocaleDateString()}`}
            </Text>
          )}
        </View>

        <View style={styles.section}>
          <Text style={styles.subtitle}>Jolocom backup service</Text>
          <Text style={styles.description}>
            Store your information using Jolocom’s backup service. With this
            option, the Jolocom team cannot view your data and your info stays
            automatically up-to-date.
          </Text>
          {isAutoBackupEnabled ? (
            <React.Fragment>
              <GradientButton
                containerStyle={styles.buttonContainer}
                textStyle={styles.buttonText}
                text={isLoading ? 'Deleting backup...' : 'Disable auto-backup'}
                onPress={onDisableAutoBackup}
                disabled={isLoading}
              />
              {lastBackup && (
                <Text style={styles.info}>
                  {`Last backup ${new Date(lastBackup).toLocaleDateString()}`}
                </Text>
              )}
            </React.Fragment>
          ) : (
            <React.Fragment>
              <GradientButton
                containerStyle={styles.buttonContainer}
                textStyle={styles.buttonText}
                onPress={enableAutoBackup}
                text={
                  isLoading ? 'Synchronization started…' : 'Use backup service'
                }
                disabled={isLoading}
              />
              <Text style={styles.warning}>
                By clicking here i agreed backing up my data on Jolocom’s
                central server and i know that it is not viewable by anyone
              </Text>
            </React.Fragment>
          )}
        </View>
      </ScrollView>
    </Container>
  )
}

export default BackupComponent
