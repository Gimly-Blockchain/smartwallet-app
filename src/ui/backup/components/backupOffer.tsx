import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { GradientButton } from '../../structure/gradientButton'
import { ActionSheet } from '../../structure/actionSheet'

interface Props {
  enableAutoBackup: () => void
  manualBackup: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  title: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 40,
  },
  logoTitle: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 108,
  },
  bottomSheet: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: Colors.black,
    height: 230,
    width: '100%',
    borderRadius: 22,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    paddingTop: 28,
    paddingHorizontal: 20,
  },
  button: {
    marginTop: 28,
    marginHorizontal: 20,
  },
  info: {
    color: Colors.white050,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 10,
  },
  textButton: {
    color: Colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
})
export const BackupOfferComponent: React.FC<Props> = ({
  enableAutoBackup,
  manualBackup,
}) => {
  return (
    <Container style={styles.container}>
      <Text style={styles.title}>
        Store your data by yourself or try to use Jolocom backup service!
      </Text>
      <View>{/* TODO add correct icon*/}</View>
      <Text style={styles.logoTitle}>Automatic synchronization</Text>
      <ActionSheet showSlide={true}>
        <GradientButton
          onPress={enableAutoBackup}
          text={'Turn on backup service'}
        />
        <Text style={styles.info}>
          By clicking here i agreed backing up my data on Jolocomâ€™s central
          server server and i know that it is not viewable by anyone
        </Text>
        <TouchableOpacity onPress={manualBackup}>
          <Text style={styles.textButton}>Keep it manually</Text>
        </TouchableOpacity>
      </ActionSheet>
    </Container>
  )
}
