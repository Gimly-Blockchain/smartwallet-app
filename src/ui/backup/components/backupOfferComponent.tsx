import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { NavigationSection } from '../../structure/navigationSection'
import { GradientButton } from '../../structure/gradientButton'

interface Props {
  enableAutoBackup: () => void
  close: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  title: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 40,
  },
  logoTitle: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 108,
  },
  bottomSheet: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: Colors.black,
    height: 230,
    width: '100%',
    borderRadius: 22,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    paddingTop: 28,
    paddingHorizontal: 20,
  },
  button: {
    marginTop: 28,
    marginHorizontal: 20,
  },
  info: {
    color: Colors.white050,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 10,
  },
  textButton: {
    color: Colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
})
const BackupOfferComponent: React.FC<Props> = ({ close, enableAutoBackup }) => {
  return (
    <Container style={styles.container}>
      <NavigationSection
        onNavigation={close}
        isBackButton={false}
        isDark={true}
      />
      <Text style={styles.title}>
        Store your data by yourself or try to use Jolocom backup service!
      </Text>
      <View>{/* TODO add correct icon*/}</View>
      <Text style={styles.logoTitle}>Automatic synchronization</Text>
      <View style={styles.bottomSheet}>
        <GradientButton
          onPress={enableAutoBackup}
          text={'Turn on backup service'}
        />
        <Text style={styles.info}>
          By clicking here i agreed backing up my data on Jolocomâ€™s central
          server and i know that it is not viewable by anyone
        </Text>
        {/*TODO add onPress*/}
        <TouchableOpacity>
          <Text style={styles.textButton}>Keep it manually</Text>
        </TouchableOpacity>
      </View>
    </Container>
  )
}

export default BackupOfferComponent
