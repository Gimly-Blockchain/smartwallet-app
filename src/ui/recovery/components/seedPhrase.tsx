import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { Container } from 'src/ui/structure'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Typography, Buttons, Spacing } from 'src/styles'

const { Button } = require('react-native-material-ui')

interface Props {
  seedPhrase: string
  handleButtonTap: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.blackMain,
  },
  noteSection: {
    marginTop: Spacing.LG,
    flex: 0.8,
    justifyContent: 'center',
  },
  note: {
    ...Typography.subMainText,
    ...Typography.centeredText,
    lineHeight: Typography.subMainText.fontSize + 2,
    color: Colors.sandLight,
  },
  phraseSection: {
    flex: 1,
  },
  seedPhrase: {
    ...Typography.baseFontStyles,
    ...Typography.centeredText,
    fontSize: Typography.text3XL,
    lineHeight: Typography.text3XL + 4,
    color: Colors.white,
  },
  buttonSection: {
    marginTop: 'auto',
  },
  buttonContainer: {
    ...Buttons.buttonStandardContainer,
  },
  buttonText: {
    ...Buttons.buttonStandardText,
  },
})

export const SeedPhrase: React.SFC<Props> = ({
  seedPhrase,
  handleButtonTap,
}: Props) => (
  <Container style={styles.container}>
    <View style={styles.noteSection}>
      <Text style={styles.note}>
        {I18n.t(strings.WRITE_THESE_WORDS_DOWN_ON_AN_ANALOG_AND_SECURE_PLACE) +
          '. ' +
          I18n.t(
            strings.WITHOUT_THESE_WORDS_YOU_CANNOT_ACCESS_YOUR_WALLET_AGAIN,
          ) +
          '.'}
      </Text>
    </View>
    <View style={styles.phraseSection}>
      <Text style={styles.seedPhrase}>{seedPhrase}</Text>
    </View>
    <View style={styles.buttonSection}>
      <Button
        style={{ container: styles.buttonContainer, text: styles.buttonText }}
        onPress={handleButtonTap}
        raised
        upperCase={false}
        text={I18n.t(strings.YES_I_WROTE_IT_DOWN)}
      />
    </View>
  </Container>
)
