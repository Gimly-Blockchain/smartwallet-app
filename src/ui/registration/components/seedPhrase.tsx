import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { Container } from 'src/ui/structure/'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'

const { Button } = require('react-native-material-ui')

interface Props {
  seedPhrase: string
  checked: boolean
  handleButtonTap: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: JolocomTheme.primaryColorBlack,
  },
  noteSection: {
    marginTop: 20,
    flex: 0.8,
    justifyContent: 'center',
  },
  note: {
    textAlign: 'center',
    lineHeight: 26,
    color: JolocomTheme.primaryColorSand,
    fontSize: JolocomTheme.labelFontSize,
    fontFamily: JolocomTheme.contentFontFamily,
  },
  phraseSection: {
    flex: 1,
  },
  seedPhrase: {
    textAlign: 'center',
    color: JolocomTheme.primaryColorWhite,
    fontSize: 34,
    fontFamily: JolocomTheme.contentFontFamily,
    lineHeight: 38,
  },
  buttonSection: {
    marginTop: 'auto',
  },
  buttonContainer: {
    borderRadius: 4,
    height: 48,
    paddingHorizontal: 25,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontWeight: '100',
    fontSize: JolocomTheme.headerFontSize,
    color: JolocomTheme.primaryColorWhite,
    fontFamily: JolocomTheme.contentFontFamily,
  },
})

export const SeedPhrase: React.SFC<Props> = props => (
  <Container style={styles.container}>
    <View style={styles.noteSection}>
      <Text style={styles.note}>
        {I18n.t(strings.WRITE_THESE_WORDS_DOWN_ON_AN_ANALOG_AND_SECURE_PLACE) +
          '. ' +
          I18n.t(
            strings.WITHOUT_THESE_WORDS_YOU_CANNOT_ACCESS_YOUR_WALLET_AGAIN,
          ) +
          '.'}
      </Text>
    </View>
    <View style={styles.phraseSection}>
      <Text style={styles.seedPhrase}>{props.seedPhrase}</Text>
    </View>
    <View style={styles.buttonSection}>
      <Button
        style={{ container: styles.buttonContainer, text: styles.buttonText }}
        onPress={props.handleButtonTap}
        raised
        upperCase={false}
        text={I18n.t(strings.YES_I_WROTE_IT_DOWN)}
      />
    </View>
  </Container>
)
