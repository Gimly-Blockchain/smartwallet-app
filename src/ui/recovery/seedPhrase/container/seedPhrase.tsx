import React from 'react'
import { connect } from 'react-redux'
import { SeedPhrase as SeedPhraseComponent } from 'src/ui/recovery/seedPhrase/components/seedPhrase'
import { ThunkDispatch } from '../../../../store'
import { NavigationScreenProps } from 'react-navigation'
import { navigationActions } from '../../../../actions'
import { routeList } from '../../../../routeList'
import { StatusBar } from 'react-native'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps>,
    NavigationScreenProps {}

export class SeedPhraseContainer extends React.Component<Props> {
  public render() {
    const mnemonic =
      this.props.navigation && this.props.navigation.getParam('mnemonic')
    return (
      <React.Fragment>
        <StatusBar
          animated={false}
          barStyle="light-content"
          showHideTransition="fade"
        />
        <SeedPhraseComponent
          seedPhrase={mnemonic}
          handleButtonTap={() => this.props.repeatSeedPhrase(mnemonic)}
        />
      </React.Fragment>
    )
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  repeatSeedPhrase: (mnemonic: string) =>
    dispatch(
      navigationActions.navigate({
        routeName: routeList.RepeatSeedPhrase,
        params: { mnemonic },
      }),
    ),
})

export const SeedPhrase = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SeedPhraseContainer)
