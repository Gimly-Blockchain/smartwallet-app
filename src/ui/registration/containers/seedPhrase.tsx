import React from 'react'
import { connect } from 'react-redux'
import { SeedPhrase as SeedPhraseComponent } from 'src/ui/registration/components/seedPhrase'
import { navigatorResetHome } from 'src/actions/navigation'
import { ThunkDispatch } from '../../../store'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {
  navigation: { state: { params: any } } // TODO Type?
}

interface State {
  checked: boolean
}

export class SeedPhraseContainer extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      checked: false,
    }
  }

  render() {
    const { mnemonic } = this.props.navigation.state.params
    return (
      <SeedPhraseComponent
        seedPhrase={mnemonic}
        checked={this.state.checked}
        handleButtonTap={this.props.finishRegistration}
      />
    )
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  finishRegistration: () => dispatch(navigatorResetHome()),
})

export const SeedPhrase = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SeedPhraseContainer)
