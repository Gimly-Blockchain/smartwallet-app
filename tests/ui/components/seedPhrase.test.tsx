import React from 'react'
import { SeedPhrase } from 'src/ui/registration/components/seedPhrase'
import { shallow } from 'enzyme'

describe('seedPhrase component', () => {
  const makeProps = (otherProps: any) => ({
    seedPhrase: 'mock seedPhrase',
    handleButtonTap: jest.fn(),
    onCheck: jest.fn(),
    ...otherProps,
  })

  it('matches the snapshot with checkbox unchecked', () => {
    const props = makeProps({ checked: false })
    const rendered = shallow(<SeedPhrase {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with checkbox checked', () => {
    const props = makeProps({ checked: true })
    const rendered = shallow(<SeedPhrase {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
