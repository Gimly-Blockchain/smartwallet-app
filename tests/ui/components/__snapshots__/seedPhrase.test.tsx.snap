// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seedPhrase component matches the snapshot with checkbox checked 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.8,
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
    >
      Write these words down on an analog and secure place. Without these words, you cannot access your wallet again.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "TT Commons",
          "fontSize": 34,
          "lineHeight": 38,
          "textAlign": "center",
        }
      }
    >
      mock seedPhrase
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
      }
    }
  >
    <ThemedComponent
      onPress={[MockFunction]}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "borderRadius": 4,
            "height": 48,
            "paddingHorizontal": 25,
          },
          "text": Object {
            "color": "white",
            "fontFamily": "TT Commons",
            "fontSize": 22,
            "fontWeight": "100",
            "paddingVertical": 15,
          },
        }
      }
      text="Yes, I wrote it down"
      upperCase={false}
    />
  </View>
</Container>
`;

exports[`seedPhrase component matches the snapshot with checkbox unchecked 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.8,
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
    >
      Write these words down on an analog and secure place. Without these words, you cannot access your wallet again.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "TT Commons",
          "fontSize": 34,
          "lineHeight": 38,
          "textAlign": "center",
        }
      }
    >
      mock seedPhrase
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
      }
    }
  >
    <ThemedComponent
      onPress={[MockFunction]}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "borderRadius": 4,
            "height": 48,
            "paddingHorizontal": 25,
          },
          "text": Object {
            "color": "white",
            "fontFamily": "TT Commons",
            "fontSize": 22,
            "fontWeight": "100",
            "paddingVertical": 15,
          },
        }
      }
      text="Yes, I wrote it down"
      upperCase={false}
    />
  </View>
</Container>
`;
